### PR TITLE
Provide basic setup for repo

### PR DESCRIPTION
**Pull request by tomkerkhove**
Closes https://github.com/arcus-azure/arcus.observability/issues/1
Closes https://github.com/arcus-azure/arcus.observability/issues/2
Closes https://github.com/arcus-azure/arcus.observability/issues/4
Closes https://github.com/arcus-azure/arcus.observability/issues/6
Closes https://github.com/arcus-azure/arcus.observability/issues/8

### CODE REVIEW

**Comment by tomkerkhove** [comment]
Builds are working:
- [CI ✔](https://dev.azure.com/codit/Arcus/_build/results?buildId=41414&view=results)
- [Release to NuGet ✔ (except release since it's another repo)](https://dev.azure.com/codit/Arcus/_build/results?buildId=41414&view=results)

I will swap the build repos in Azure DevOps after merging.

File: src/Arcus.Observability.Correlation/CorrelationInfo.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using GuardNet;
+
+namespace Arcus.Observability.Correlation
+{
+    /// <summary>
+    ///     Represents the correlation ID information on the incoming requests and outgoing responses.
+    /// </summary>
+    public class CorrelationInfo
```
Comment (tomkerkhove) [inline src/Arcus.Observability.Correlation/CorrelationInfo.cs]: This will be the new home for this one @stijnmoreels

Comment (tomkerkhove) [reply src/Arcus.Observability.Correlation/CorrelationInfo.cs]: Then Web API & Messaging can use this and extend it if required.

Comment (stijnmoreels) [reply src/Arcus.Observability.Correlation/CorrelationInfo.cs]: Yes, merged.

**Pull request merged**